### PR TITLE
Improve how connections are rejected after shutdown

### PR DESCRIPTION
**Pull request by AthenaEryma**
Prior to this PR, when a channel was accepted after Elasticsearch received a shutdown signal, `AbstractHttpServerTransport` would directly close the connection. This PR makes it throw an exception instead, which results in the connection being closed by Netty. This keeps the flow of control more consistent and readable, as well as logging additional details such as the host which made the request.

This PR doesn't include any test changes because the behavior doesn't really change in a way that's easy to test.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java [hunk 1]
```diff
@@ -388,9 +388,7 @@ protected void serverAcceptedChannel(HttpChannel httpChannel) {
         try {
             rlock.lock();
             if (shuttingDown) {
-                logger.warn("server accepted channel after shutting down");
-                httpChannel.close();
-                return;
+                throw new IllegalStateException("server accepted channel after shutting down");
```
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java:391]: The error message is a bit cryptic, could we improve it? Something like `Cannot accept new server channel while shutting down`

**Review by rjernst — APPROVED** [review]
LGTM, one suggestion

**Review by Tim-Brooks — COMMENTED** [review]
This change seems fine to me.

**Review by stu-elastic — APPROVED** [review]
Much appreciated, nicer to let the caller handle the channel.

File: server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java (hunk 1 above)
Comment (gwbrown) [reply server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java]: Good call, done.

**Pull request merged**